pkg/executor/delete.go

```diff
@@ -128,6 +132,19 @@ func (e *DeleteExec) deleteSingleTableByChunk(ctx context.Context) error {
 				datumRow = append(datumRow, datum)
 			}
 
+			if e.ignoreErr {
+				fkToBeCheckedRows[0] = toBeCheckedRow{row: datumRow, ignored: false}
+				err := checkFKIgnoreErr(ctx, e.Ctx(), e.fkChecks[tbl.Meta().ID], fkToBeCheckedRows)
+				if err != nil {
+					return err
+				}
+
+				// meets an error, skip this row.
+				if fkToBeCheckedRows[0].ignored {
+					datumRow = datumRow[:0]
+					continue
+				}
+			}
```

Comment:
How about move this into `removeRow`? 
this is duplicate with #251~#263

Reply:
I prefer to keep it outside. If we move it inside the `removeRow`, we'll need to handle the error carefully to distinguish the error, the ignored error and success.

Instead, I change the implementation of `checkFKIgnoreErr` to integrate the declaration and usage of the `fkToBeCheckedRows[0]` usage, and make it easier to understand / use:

```go
func checkFKIgnoreErr(ctx context.Context, sctx sessionctx.Context, fkChecks []*FKCheckExec, row []types.Datum) (bool, error)
```

Now, only the `row` is needed, and it'll return whether an error has been ignored, or there is an unignorable error. The usage can be simpler:

```go
			if e.ignoreErr {
				ignored, err := checkFKIgnoreErr(ctx, e.Ctx(), e.fkChecks[tbl.Meta().ID], datumRow)
				if err != nil {
					return err
				}

				// meets an error, skip this row.
				if ignored {
					datumRow = datumRow[:0]
					continue
				}
			}
```